### DISCUSSION TITLE
Unable to Delete Questionsets

### DISCUSSION THREAD

**Discussion by siddhishinde0723** [Sunbird-Ed/Community#699] [General Discussion]
Hello Team,

We are experiencing difficulties with deleting Questionsets on Sunbird SaaS Release 5.0. Despite repeated attempts, the deletion process fails to remove the selected Questionsets from the platform.

Issue Details:
When attempting to delete a Questionset, the platform does not execute the deletion process successfully. This issue persists even after multiple attempts.
![image](https://github.com/Sunbird-Ed/Community/assets/72926048/00888b0f-360e-449b-ad02-96095bb3079b)

I have alos check on staging as well getting the same issue.
![image](https://github.com/Sunbird-Ed/Community/assets/72926048/c46ef90b-b4f1-4f8a-9dff-cde080ec8449)

Has anyone encountered the same issue? Is there any solution for the same?
Please provide guidance or assistance if you have encountered a similar issue or have any suggestions for resolving it.

@vinukumar-vs @rajeevsathish @rajeshkumaravel @Krishnaj20

**Comment by ashokreddy1208** [comment]
@siddhishinde0723 We have a known bug for question set delete functionality, please find issue details here - https://project-sunbird.atlassian.net/browse/CO-482

Any how we are able to delete question set from API, please find screen shot for your reference.
![image (16)](https://github.com/Sunbird-Ed/Community/assets/58033030/75d21815-27ba-492e-9851-7a2b5025af80)
![image (5)](https://github.com/Sunbird-Ed/Community/assets/58033030/50c01d14-ed56-4d54-b7c9-7489960b77db)
![image (4)](https://github.com/Sunbird-Ed/Community/assets/58033030/59c4ad03-e3d8-46df-8b1b-3dd793592a54)


cc : @vinukumar-vs @rajeevsathish @rajeshkumaravel @Krishnaj20

**Reply by siddhishinde0723** [reply]
@ashokreddy1208 We appreciate your advice; we can now use the API to remove the questionset. 
Is this issue solved? Or in which release can this issue be solved?

**Comment by ashokreddy1208** [comment]
@siddhishinde0723 deleting question set from API is supported from long time, you can use any of the below curls to delete question set.

`curl --location --globoff --request DELETE '[https://staging.sunbirded.org/api/questionset/v2/retire/{{questionset_id}}](https://staging.sunbirded.org/api/questionset/v2/retire/%7B%7Bquestionset_id%7D%7D)' \
--header 'Authorization: Bearer {{bearer_token}}' \
--header 'X-Channel-Id: 0138934136407244800' \
--header 'Content-Type: application/json' \
--data '{
    "request": {
        "questionset": {}
    }
}'`


`curl --location --request DELETE 'https://staging.sunbirded.org/api/content/v2/retire/{{questionset_id}}' \
--header 'X-Authenticated-User-token: {{authToken}}' \
--header 'Authorization: {{APIKey}}' \
--header 'Accept: application/json'`

**Reply by siddhishinde0723** [reply]
@ashokreddy1208 Thanks for the guidance.
